### PR TITLE
Debianrepochecker: also get arch from flathub.json

### PR DESCRIPTION
**Pull request by Lctrs**
If only one arch is defined in flathub.json, use this one.

Following comment made by @barthalion in https://github.com/flathub/flathub/pull/1292#discussion_r357015490

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -183,6 +182,14 @@ def open_pr(subject, body, branch):
     log.info("Opened pull request %s", pr.html_url)
 
 
+def load_flathubconfig(path):
+    if not os.path.isfile(path + "/flathub.json"):
+        return None
+
+    with open(path + "/flathub.json") as f:
+        return json.load(f)
```
Comment (wjt) [inline src/main.py]: How about:

```python3
try:
    with open(os.path.join(path, "flathub.json"), "r") as f:
        return json.load(f)
except FileNotFoundError:
    return {}
```

**Comment by wjt** [comment]
I've got to say that I don't agree with the comment at https://github.com/flathub/flathub/pull/1292#discussion_r357015490 – I think it's correct for the manifest to be self-contained and not be reliant on information from `flathub.json` – but nonetheless this feature seems useful since evidently others disagree :-)

File: src/main.py (hunk 1 above)
Comment (Lctrs) [reply src/main.py]: Adressed.

**Comment by Lctrs** [comment]
I have no strong opinion on it. Let's wait for @barthalion input.

**Comment by barthalion** [comment]
Yeah, I don't think using flathub.json is fine either. I'd rather add `arch` field into `x-checker-data` and keep `only-arches` as a fallback only if it's unavailable here. (Yes, I'm not a fan of mixing flatpak-builder and external tooling metadata.)

**Comment by Lctrs** [comment]
Okay. Closing then.

**Pull request closed without merging**